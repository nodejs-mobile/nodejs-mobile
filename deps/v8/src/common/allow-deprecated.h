// Copyright 2022 the V8 project authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#ifndef V8_COMMON_ALLOW_DEPRECATED_H_
#define V8_COMMON_ALLOW_DEPRECATED_H_

#if defined(V8_IMMINENT_DEPRECATION_WARNINGS) || \
    defined(V8_DEPRECATION_WARNINGS)

#if defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  __pragma(warning(push)) __pragma(warning(disable : 4996))

#define END_ALLOW_USE_DEPRECATED() __pragma(warning(pop))

#else  // !defined(V8_CC_MSVC)

#define START_ALLOW_USE_DEPRECATED() \
  _Pragma("GCC diagnostic push")     \
      _Pragma("GCC diagnostic ignored \"-Wdeprecated-declarations\"")

#define END_ALLOW_USE_DEPRECATED() _Pragma("GCC diagnostic pop")

#endif  // !defined(V8_CC_MSVC)

#else  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
       // defined(V8_DEPRECATION_WARNINGS))

#define START_ALLOW_USE_DEPRECATED()
#define END_ALLOW_USE_DEPRECATED()

#endif  // !(defined(V8_IMMINENT_DEPRECATION_WARNINGS) ||
        // defined(V8_DEPRECATION_WARNINGS))

#endif  // V8_COMMON_ALLOW_DEPRECATED_H_
