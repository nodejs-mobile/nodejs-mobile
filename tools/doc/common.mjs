import yaml from 'js-yaml';

export function isYAMLBlock(text) {
  return /^<!-- YAML/.test(text);
}

export function isSourceLink(text) {
  return /^<!-- source_link=([^\s/]+\/)+\w+\.\w+ -->/.test(text);
}

export function arrify(value) {
  return Array.isArray(value) ? value : [value];
}

export function extractAndParseYAML(text) {
  text = text.trim()
             .replace(/^<!-- YAML/, '')
             .replace(/-->$/, '');

  // js-yaml.load() throws on error.
  const meta = yaml.load(text);

  if (meta.added) {
    // Since semver-minors can trickle down to previous major versions,
    // features may have been added in multiple versions.
    meta.added = arrify(meta.added);
  }

  if (meta.napiVersion) {
    meta.napiVersion = arrify(meta.napiVersion);
  }

  if (meta.deprecated) {
    // Treat deprecated like added for consistency.
    meta.deprecated = arrify(meta.deprecated);
  }

  if (meta.removed) {
    meta.removed = arrify(meta.removed);
  }

  meta.changes = meta.changes || [];

  return meta;
}
