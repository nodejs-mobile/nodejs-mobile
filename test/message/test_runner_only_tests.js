// Flags: --no-warnings --test-only
'use strict';
require('../common');
const test = require('node:test');

// These tests should be skipped based on the 'only' option.
test('only = undefined');
test('only = undefined, skip = string', { skip: 'skip message' });
test('only = undefined, skip = true', { skip: true });
test('only = undefined, skip = false', { skip: false });
test('only = false', { only: false });
test('only = false, skip = string', { only: false, skip: 'skip message' });
test('only = false, skip = true', { only: false, skip: true });
test('only = false, skip = false', { only: false, skip: false });

// These tests should be skipped based on the 'skip' option.
test('only = true, skip = string', { only: true, skip: 'skip message' });
test('only = true, skip = true', { only: true, skip: true });

// An 'only' test with subtests.
test('only = true, with subtests', { only: true }, async (t) => {
  // These subtests should run.
  await t.test('running subtest 1');
  await t.test('running subtest 2');

  // Switch the context to only execute 'only' tests.
  t.runOnly(true);
  await t.test('skipped subtest 1');
  await t.test('skipped subtest 2');
  await t.test('running subtest 3', { only: true });

  // Switch the context back to execute all tests.
  t.runOnly(false);
  await t.test('running subtest 4', async (t) => {
    // These subtests should run.
    await t.test('running sub-subtest 1');
    await t.test('running sub-subtest 2');

    // Switch the context to only execute 'only' tests.
    t.runOnly(true);
    await t.test('skipped sub-subtest 1');
    await t.test('skipped sub-subtest 2');
  });

  // Explicitly do not run these tests.
  await t.test('skipped subtest 3', { only: false });
  await t.test('skipped subtest 4', { skip: true });
});
