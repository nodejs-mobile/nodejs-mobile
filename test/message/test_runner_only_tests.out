TAP version 13
# Subtest: only = undefined
ok 1 - only = undefined # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = undefined, skip = string
ok 2 - only = undefined, skip = string # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = undefined, skip = true
ok 3 - only = undefined, skip = true # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = undefined, skip = false
ok 4 - only = undefined, skip = false # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = false
ok 5 - only = false # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = false, skip = string
ok 6 - only = false, skip = string # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = false, skip = true
ok 7 - only = false, skip = true # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = false, skip = false
ok 8 - only = false, skip = false # SKIP 'only' option not set
  ---
  duration_ms: *
  ...
# Subtest: only = true, skip = string
ok 9 - only = true, skip = string # SKIP skip message
  ---
  duration_ms: *
  ...
# Subtest: only = true, skip = true
ok 10 - only = true, skip = true # SKIP
  ---
  duration_ms: *
  ...
# Subtest: only = true, with subtests
    # Subtest: running subtest 1
    ok 1 - running subtest 1
      ---
      duration_ms: *
      ...
    # Subtest: running subtest 2
    ok 2 - running subtest 2
      ---
      duration_ms: *
      ...
    # Subtest: skipped subtest 1
    ok 3 - skipped subtest 1 # SKIP 'only' option not set
      ---
      duration_ms: *
      ...
    # Subtest: skipped subtest 2
    ok 4 - skipped subtest 2 # SKIP 'only' option not set
      ---
      duration_ms: *
      ...
    # Subtest: running subtest 3
    ok 5 - running subtest 3
      ---
      duration_ms: *
      ...
    # Subtest: running subtest 4
        # Subtest: running sub-subtest 1
        ok 1 - running sub-subtest 1
          ---
          duration_ms: *
          ...
        # Subtest: running sub-subtest 2
        ok 2 - running sub-subtest 2
          ---
          duration_ms: *
          ...
        # Subtest: skipped sub-subtest 1
        ok 3 - skipped sub-subtest 1 # SKIP 'only' option not set
          ---
          duration_ms: *
          ...
        # Subtest: skipped sub-subtest 2
        ok 4 - skipped sub-subtest 2 # SKIP 'only' option not set
          ---
          duration_ms: *
          ...
        1..4
    ok 6 - running subtest 4
      ---
      duration_ms: *
      ...
    # Subtest: skipped subtest 3
    ok 7 - skipped subtest 3 # SKIP 'only' option not set
      ---
      duration_ms: *
      ...
    # Subtest: skipped subtest 4
    ok 8 - skipped subtest 4 # SKIP
      ---
      duration_ms: *
      ...
    1..8
ok 11 - only = true, with subtests
  ---
  duration_ms: *
  ...
1..11
# tests 11
# pass 1
# fail 0
# cancelled 0
# skipped 10
# todo 0
# duration_ms *
